src/Farfetch.LoadShedding.Prometheus/Metrics/HttpRequestsConcurrencyItemsGauge.cs

```diff
@@ -22,16 +21,27 @@ public class HttpRequestsConcurrencyItemsGauge : MetricBase<Gauge>
         protected override string DefaultName => "http_requests_concurrency_items_total";
 
         /// <summary>
-        /// Sets the value of the gauge.
+        /// Increments the value of the gauge.
+        /// </summary>
+        /// <param name="method">The method.</param>
+        /// <param name="priority">The priority.</param>
+        public void Increment(string method, string priority)
+        {
+            this.Metric?
+                .WithLabels(method, priority)
+                .Inc();
+        }
+
+        /// <summary>
+        /// Decrements the value of the gauge.
         /// </summary>
         /// <param name="method">The method.</param>
         /// <param name="priority">The priority.</param>
-        /// <param name="value">The value.</param>
-        public void Set(string method, string priority, double value)
+        public void Decrement(string method, string priority)
```

Comment:
This change introduces a breaking change, if, for some reason, someone calls the Set method, right?

The same applies to other changes from a Set method to an Increment and Decrement methods.

Reply:
Hi @brmagadutra , The class and method are public, so I would say, Yes, it's a breaking change. However, I think everyone should call the extension `IAdaptativeLimiterOptionsExtensions.AddMetrics` to configure the `LoadSheddingOptions`. It will have no impact on who use this extension method that configures the HttpRequestsConcurrencyItemsGauge.

Reply:
I understand, and this would be the best way. Still, we have this scenario to consider, so if the rest of the team is ok I won't oppose going forward. @joelfoliveira and @ailtonguitar what do you think?